crypto/evp/m_sigver.c

```diff
@@ -196,6 +200,9 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
      */
     ERR_pop_to_mark();
 
+    if (type == NULL && mdname != NULL)
```

Comment:
Why is this legacy path being hit by TLS?

Reply:
In `tls_psk_do_binder` we have to create an HMAC:

https://github.com/openssl/openssl/blob/8658feddea6aef5cf5cbb1cfbf6b1817fa432051/ssl/statem/extensions.c#L1597-L1605

We do so using EVP_DigestSignInit_ex() and pass the name of the digest to be used. In the test I was looking at that failed it was SHA384. Because we're actually creating a MAC and there are no "signature" implementations in the providers for MACs, we go down the legacy path and end up in crypto/evp/pkey_mac.c. This does the work of converting the call to EVP_MAC - and so eventually we will end up in a provider, via a legacy EVP_PKEY_METHOD.

Without this line type is NULL (because we actually called EVP_DigestSignInit_ex() and provided an mdname). This then resulted in us using the default digest for HMAC which is SHA256. Thus everything fell over because the MAC we generated wasn't the expected size.